### ISSUE TITLE
PhantomJS crashes when running chutzpah.console.exe

### ISSUE THREAD

**Issue by mcweiss** [mmanela/chutzpah#389]
I get the following error:
***Error: Unknown error occurred when executing test file. Received exit code of -1073741819
While Running:***C:\...one-of-mytestfiles.js

This happens often, but not every time, seems worse on the build machine - but it does happen on my workstation as well.

If I capture the temporary html file the console app creates and the command line it uses to execute PhantomJS (sysinternal's process monitor and process explorer are *the best*) I can recreate this outside the your console application.

PhantomJS says this:

PhantomJS has crashed. Please read the crash reporting guide at https://github.com/ariya/phantomjs/wiki/Crash-Reporting and file a bug report at https
://github.com/ariya/phantomjs/issues/new with the crash dump file attached: <path to a dump file>

Without symbols the dump was pretty useless.

I can, however, pinpoint the test the fails, but it's nearly identical to other similar tests in that file, so I can't see what I could be doing "wrong" enough to crash the browser. This test, the tests in that file, and the tests in other files all run file in Visual Studio using your tool. They also run fine in the Jasmine test runner page we also use. 

**While this isn't really a Chutzpah issue, but I have a few requests:**

**1. I'd like to see if PhantomJS 2.0 fixes this.** You're using 1.9. Dropping the new version into the install directory doesn't work. The first issue was chutzpahRunner.js doesn't work. See this:

http://phantomjs.org/api/phantom/property/args.html

I fixed that issue, but I still can get it to work. Now I see this:

#_#Error#_# {"type":"Error","error":{"message":"Protocol \"e\" is unknown"}}
#_#Log#_# {"type":"Log","log":{"message":"!!_!! Resource Error: e:%5Csrc%5CPRODUCTVSO%5CMain%5CMY PRODUCT%5CSource%5CPRODUCT%5Cpackages%5CChutzpah.3.3.1%5Ctools%5CJSRunners%5CjasmineRunnerV2.js"}}
#_#Log#_# {"type":"Log","log":{"message":"!!_!! Resource Recieved: e:%5Csrc%5CPRODUCTVSO%5CMain%5CMY%20PRODUCT%5CSource%5CPRODUCT%5Cpackages%5CChutzpah.3.3.1%5Ctools%5CJSRunners%5CjasmineRunnerV2.js"}}
#_#Log#_# {"type":"Log","log":{"message":"!!_!! onInitialized  Jasmine - v2"}}
#_#Log#_# {"type":"Log","log":{"message":"!!_!! isJasmineLoaded"}}

**Have you investigated using PhantomJS 2.0?** Any suggestions on how to make this work to see if 2.0 fixes the problem?

**2. Logging the current  test executing after a crash**
It would have been nice if the the console application logged the currently running test when PhantomJS crashed. I can see this "#_#TestStart#_#" {...} right before it crashes so I would think this is possible. Also I see an error message from PhantomJS  written on out to stderr, why not include that in the log too?

**Comment by mmanela** [comment]
1. The current version of Chutzpah uses Phantom 2.0. What version are you using?

2. If you use the /trace flag you should see all the info Chutzpah knows about. Are you suggesting it should always print the test it is running to the console?

**Comment by mcweiss** [comment]
Thanks for the fast reply.

It appears I haven't updated the CONSOLE version of Chtuzpah, this is installed separately from the VS add-in it seems. The console version was still at 3.3.1.0. I guess I forgot there's a difference between the add-in and the Solution's NuGet packages...

I was able to "fix" my older version to run with PhantomJS 2.0 after I submitted this issue (I was one-off in System.args) . Once it ran with the newer browser my issue seems to go away. So forget my request to update to PhantonJS 2.0, seems you have. I'll just have to update the solution's package. This should then fix our build machines and developer's machines hopefully.

My other suggestion was around this logging:

    Chutzpah console test runner  (64-bit .NET 4.0.30319.34209)
    Version 3.3.1.0
    Copyright (C) 2013 Matthew Manela (http://matthewmanela.com).
    
    chutzpah.dll:     Version 3.3.1.0
    
    Error: Unknown error occurred when executing test file. Received exit code of -1073741819
    While Running:E:\src\blah\blah\my.tests.js
    
    
    === 10 total, 0 failed, took 3.21 seconds ===

It would have been nice if I could have seen which test was being run when PhantomJS crashed. When running PhantomJS myself on the command line I could see the stream of logging text your console application must capture. However, yes I suppose /debug would have shown me this. I guess that flag would be good enough on the extremely rare case when PhantomJS crashes. 

At the very least I've learned quite a bit about how the console version works! Thanks for a great tool!
